presto-main/src/main/java/io/prestosql/sql/planner/optimizations/BeginTableWrite.java

```diff
@@ -187,6 +187,16 @@ private WriterTarget createWriterTarget(WriterTarget target)
                 DeleteTarget delete = (DeleteTarget) target;
                 return new DeleteTarget(metadata.beginDelete(session, delete.getHandle()), delete.getSchemaTableName());
             }
+            if (target instanceof TableWriterNode.RefreshMaterializedViewReference) {
+                TableWriterNode.RefreshMaterializedViewReference refreshMV = (TableWriterNode.RefreshMaterializedViewReference) target;
+                boolean skipRefresh = false;
+
+                if (metadata.isMaterializedViewCurrent(session, refreshMV.getMaterializedViewHandle()).getIsFresh()) {
+                    skipRefresh = true;
+                }
+                return new TableWriterNode.RefreshMaterializedViewTarget(metadata.beginRefreshMaterializedView(session, refreshMV.getStorageTableHandle(), skipRefresh),
```

Comment:
As we discussed offline, the freshness check and short-circuiting should be done in the RefreshMaterializedView task class, before we even get to the planner/optimizer.

Reply:
We discussed offline. In statement analysis we figure out whether refresh should be skipped and if yes, planner plans a no-op plan.